tests/test_search.py

```diff
@@ -2284,3 +2328,16 @@ def test_geoshape(client: redis.Redis):
     assert result.docs[0]["id"] == "small"
     result = client.ft().search(q2, query_params=qp2)
     assert len(result.docs) == 2
+
+
+def _assert_geosearch_result(client, result, expected_doc_ids):
+    """
+    Make sure the result of a geo search is as expected, taking into account the RESP
+    version being used.
+    """
+    if is_resp2_connection(client):
+        assert set([doc.id for doc in result.docs]) == set(expected_doc_ids)
+        assert result.total == len(expected_doc_ids)
+    else:
+        assert set([doc["id"] for doc in result["results"]]) == set(expected_doc_ids)
+        assert result["total_results"] == len(expected_doc_ids)
```

Comment:
So for RESP2 the response type is an object and for RESP3 array?

Reply:
While checking this, I found one existing test that was not asserting well for RESP3, I updated it. Most probably the test does not even run with RESP3, since there were no failures in the past. A thing to check, but I'll leave that for a separate PR.